relay-modern/src/lib/isomorphic-relay/PreloadLink.js

```diff
@@ -1,36 +1,69 @@
-import React, { Component, PropTypes } from 'react'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
 import { Link } from 'react-router-dom'
+import { getRelayRouteProps } from './getRelayRouteProps'
+import { routes } from 'apps/artworks/routes'
+import { fetchQuery } from 'react-relay'
+import { getRelayEnvironment } from './getRelayEnvironment'
+import * as cache from './cache'
 
-class PreloaderLink extends Component {
+export class PreloadLink extends Component {
   static contextTypes = {
     router: PropTypes.shape({
-      replace: PropTypes.func.isRequired,
-      push: PropTypes.func.isRequired,
+      history: PropTypes.shape({
+        replace: PropTypes.func.isRequired,
+        push: PropTypes.func.isRequired,
+      }),
     }).isRequired,
   }
 
   state = {
     isLoading: false,
   }
 
+  fetchData() {
+    return new Promise(async (resolve, reject) => {
+      // TODO: Check cache against relay store
+      const cacheKey = this.props.to
+
+      if (cache.get(cacheKey)) {
+        return resolve()
+      }
+
+      const { query, variables } = getRelayRouteProps(routes, cacheKey)
+      const environment = getRelayEnvironment()
+
+      try {
+        if (query) {
+          const response = await fetchQuery(environment, query, variables)
+          cache.set(cacheKey, response)
```

Comment:
@alloy - I'll ping you about this on Monday, but I'm trying to think of a good way to cleanly and accurately get records from the store. I've been looking through [these docs](https://facebook.github.io/relay/docs/en/relay-store.html) but its not entirely clear... With access to the query, is it possible to say something along the lines of `record = store.get(query)`? How does one identify the cacheId? (By "query" I mean something like a stringified version [of this](https://github.com/damassi/isomorphic-relay-app/blob/master/relay-modern/src/apps/artworks/routes/artists/ArtistsRoute.js#L12)). 

Line 39 is just a make-your-own-dumb-key-value-map so that i can get the functionality right for this 
```javascript
<PreloadLink to='/some/route'>
  Click me, page will transition after done loading relay data, and then cache...
</PreloadLink>
``` 
component, and it currently works, but I would like to use Relay's if possible so I don't have to manage my own cache. Any thoughts would be appreciated.

Reply:
Relay Modern does cache denormalisation by `Node` ID. _If_ you’re using our fork that inflects that from our schema, then it would be the value of `__id`. If you’re currently _not_ using that fork, then the ID would `jean-michel-basquiat`, but that’s clearly not a GID so while it will work in simple examples it’s not a reliable situation.